ci/docker/runtime_functions.sh

```diff
@@ -33,35 +33,23 @@ clean_repo() {
     git submodule update --init --recursive
 }
 
-build_ccache_wrappers() {
-    set -ex
-
-    rm -f cc
-    rm -f cxx
-
-    touch cc
-    touch cxx
-
-    if [ -z ${CC+x} ]; then
-        echo "No \$CC set, defaulting to gcc";
-        export CC=gcc
-    fi
-
-    if [ -z ${CXX+x} ]; then
-       echo "No \$CXX set, defaulting to g++";
-       export CXX=g++
-    fi
-
-    # this function is nessesary for cuda enabled make based builds, since nvcc needs just an executable for -ccbin
 
-    echo -e "#!/bin/sh\n/usr/local/bin/ccache ${CC} \"\$@\"\n" >> cc
-    echo -e "#!/bin/sh\n/usr/local/bin/ccache ${CXX} \"\$@\"\n" >> cxx
+install_ccache_wrappers() {
+    set -ex
 
-    chmod +x cc
-    chmod +x cxx
+    # Add to the beginning of path to ensure this redirection is picked up instead
+    # of the original ones. Especially CUDA/NVCC appends itself to the beginning of the
+    # path and thus this redirect is ignored. This change fixes this problem
+    export PATH=/usr/local/bin:$PATH
+    ln -s ccache /usr/local/bin/gcc
```

Comment:
Would this work if already set?

Reply:
Yes, you can define and configure the compilers ($CC, $CXX, $NVCC) as you want, we only intercept the backend calls to the actual compilers. This allows us to be independent of the configuration.

Reply:
I meant if ccache soft link is already there.

Reply:
In that case, the softlink (or whatever was there previously) would just get replaced. But since this script is executed at the install time of the container, it's highly unlikely that the link already exists